src/timeline/Timeline.tsx

```diff
@@ -107,7 +121,8 @@ const Timeline = (props: TimelineProps) => {
     start = 0,
     end = 24,
     date,
-    events = [],
+    pageEvents = [[], [], [], [], [], [], []],
+    pageDates = [],
```

Comment:
Notice that we already have the `date` prop, maybe we should unify it with `pageDates` 
Meaning, we can change `date` type to be either `string` or `string[]` and it can serve two purposed. 
I think it's better than having two different props with a similar purpose

Another option is use `date` and `numberOfDays` to calculate the dates array yourself.

Reducing props can make the API more clear and prune to misuse